experimenter/experimenter/experiments/models.py

```diff
@@ -793,33 +793,54 @@ def computed_duration_days(self):
         return self.proposed_duration
 
     def timeline(self):
-        return [
+        timeline_entries = [
             {
                 "label": self.Status.DRAFT,
                 "date": self.draft_date,
                 "is_active": self.is_draft,
+                "days": None,
             },
             {
                 "label": self.Status.PREVIEW,
                 "date": self.preview_date,
                 "is_active": self.is_preview,
+                "days": None,
             },
             {
                 "label": self.PublishStatus.REVIEW,
                 "date": self.review_date,
                 "is_active": self.is_review,
+                "days": None,
             },
             {
                 "label": self.Status.LIVE,
                 "date": self.start_date,
                 "is_active": self.is_live,
+                "days": None,
             },
             {
                 "label": self.Status.COMPLETE,
                 "date": self.computed_end_date,
                 "is_active": self.is_complete,
+                "days": self.computed_duration_days if self.computed_end_date else None,
             },
         ]
+        if not self.is_rollout:
+            timeline_entries.insert(
+                4,
+                {
+                    "label": NimbusConstants.ENROLLMENT_END,
+                    "date": self._enrollment_end_date,
+                    "is_active": self._enrollment_end_date is not None,
+                    "days": (
+                        self.computed_enrollment_days
+                        if self._enrollment_end_date
+                        else None
+                    ),
```

Comment:
This has a couple things I'm unsure about (as is often the case for experiment dates...):

1. I'm not quite sure if the logic is correct here: `self.computed_enrollment_days` is a computed property, and if `self._enrollment_end_date` is not set but `self.is_paused == True`, then it goes and computes and sets `self._enrollment_end_date` based on the changelogs. Maybe we should change the `if` here to include `or self.is_paused`? Not sure if that's only a relic of Normandy or if it's still used.
2. I think the `if` should include `is not None` like you have above for `is_active` (though not sure if this is likely to have an effect in practice).

Reply:
I changed the logic now, would you like to take another look into that?